src/Workspaces/Core/Portable/Utilities/CancellationTokenExtensions.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Shared.Utilities
+{
+    /// <summary>
+    /// Extensions to <see cref="CancellationToken"/>.
+    /// </summary>
+    internal static class CancellationTokenExtensions
```

Comment:
📝 Taken from Microsoft/vs-threading. This class avoids creation of a linked token source when one of the inputs is not cancellable. In Roslyn production runs, this is _always_ the case, so the behavior provided by `CombineWith` is required for preventing some amount of performance regression in the asynchronous processing queues.

Reply:
Type has been removed.